include/oneapi/dpl/pstl/glue_algorithm_ranges_defs.h

```diff
@@ -125,6 +145,18 @@ template <typename _ExecutionPolicy, typename _Range>
 oneapi::dpl::__internal::__enable_if_execution_policy<_ExecutionPolicy, bool>
 is_sorted(_ExecutionPolicy&& __exec, _Range&& __rng);
 
+// [alg.merge]
+
+template <typename _ExecutionPolicy, typename _Range1, typename _Range2, typename _Range3, typename _Compare>
+oneapi::dpl::__internal::__enable_if_execution_policy<_ExecutionPolicy,
+                                                      oneapi::dpl::__internal::__difference_t<_Range3>>
```

Comment:
Range based API algo return  a difference type instead of OutputIterator, to avoid introduce type of "OutputIterator", which is compatible with a corresponding range.
Actually, there are no  parallel versions of range based API algorithm in the standard yet. Currently we have added a kind of extensions.